C/sort/counting sort.c

```diff
@@ -0,0 +1,61 @@
+/* Counting sort is a sorting algorithm that sorts the elements of an array by 
+counting the number of occurrences of each unique element in the array. 
+The count is stored in an auxiliary array and the sorting is done by 
+mapping the count as an index of the 
+auxiliary array.
+# The average time complexity for Counting Sort is O(n + k). 
+  The worst time complexity is O(n + k).
+# The best complexity for Counting Sort is O(n + k). */
+
+/* C Program to sort Elements Using Counting Sort */
+#include <stdio.h>  
+ 
+/*  Counting sort function  */
+void counting_sort(int A[], int k, int n)
+{
+    int i, j;
+    int B[15], C[100];
+    for (i = 0; i <= k; i++)
+        C[i] = 0;
+    for (j = 1; j <= n; j++)
+        C[A[j]] = C[A[j]] + 1;
+    for (i = 1; i <= k; i++)
+        C[i] = C[i] + C[i-1];
+    for (j = n; j >= 1; j--)
+    {
+        B[C[A[j]]] = A[j];
+        C[A[j]] = C[A[j]] - 1;
+    }
+    printf("The Sorted array is : ");
+    for (i = 1; i <= n; i++)
+        printf("%d ", B[i]);
+}
+/*  End of counting_sort()  */
+ 
+/*  The main() begins  */
+int main()
+{
+    int n, k = 0, A[15], i;
```

Comment:
Do not fix the size of the array, declare the array as ```A[n];``` after taking the size of the array as input from the user.